command/agent/http.go

```diff
@@ -277,6 +277,11 @@ func handleUI(h http.Handler) http.Handler {
 
 func handleRootRedirect() http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		if req.URL.Path != "/" {
+			w.WriteHeader(http.StatusNotFound)
+			return
+		}
+
 		http.Redirect(w, req, "/ui/", 307)
 		return
```

Comment:
very picky style nit pick - I'd consider swapping conditionals, such that we redirect inside the if clause as it's the special case; and the last return is meaningless - I find the following reads a tad better.

```go
	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
		if req.URL.Path == "/" {
			http.Redirect(w, req, "/ui/", 307)
		} else {
			w.WriteHeader(http.StatusNotFound)
		}
	})
```

Reply:
I like this better too, thanks